### PR TITLE
Shop

### LINKED ISSUE #131: Shop page
We must have a shop where users are able to trade their points for goodies.

### PR DESCRIPTION
**Pull request by toni-santos**
Closes #131

### CODE REVIEW

File: resources/js/Components/Shop/ShopItem.vue [hunk 1]
```diff
@@ -0,0 +1,25 @@
+<script setup lang="ts">
+import { defineProps } from 'vue';
+
+interface Props {
+    name: string;
+    price: number;
+    description: string;
+    image: string;
+}
```
Comment (Naapperas) [inline resources/js/Components/Shop/ShopItem.vue]: We already have the Product type: you should use that, even when mocking.

File: resources/js/Pages/Shop.vue [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    name: 'Item 1',
+    price: 100,
+    description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.',
+    image: 'https://picsum.photos/200/200'
+},{
+    name: 'Item 1',
+    price: 100,
+    description: 'Lorem ipsum dolor sit amet consectetur adipisicing elit. Quisquam, voluptatum.',
+    image: 'https://picsum.photos/200/200'
+},]
```
Comment (Naapperas) [inline resources/js/Pages/Shop.vue]: Same as above.

**Review by Naapperas — CHANGES_REQUESTED** [review]

File: app/Http/Controllers/ShopController.php [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+{
+    public function show(Request $request)
+    {
+        $edition = $request->input('edition');
+
+        if ($edition === null) {
+            return response('No edition found', 500);
+        }
+
+        $products = Product::where('edition_id', $edition->id)->get();
```
Comment (Naapperas) [inline app/Http/Controllers/ShopController.php]:
```suggestion
        $products = $edition->products;
```

This is more idiomatic, in my opinion.

Comment (Naapperas) [reply app/Http/Controllers/ShopController.php]: You could even inline this in the return value

File: resources/js/Components/Shop/ShopItem.vue
Comment (Naapperas) [inline resources/js/Components/Shop/ShopItem.vue:1]: Are you going to implement buying items? Nonetheless, we should already have a button for that. If you have other ideas please share them.

**Review by Naapperas — CHANGES_REQUESTED** [review]
There is no way for a user to go to the shop unless they know the exact Url, maybe a link in the Navbar?

**Review by Naapperas — DISMISSED** [review]

**Comment by pedronunes19** [comment]
> There is no way for a user to go to the shop unless they know the exact Url, maybe a link in the Navbar?

Was this fixed?

**Comment by Naapperas** [comment]
> > There is no way for a user to go to the shop unless they know the exact Url, maybe a link in the Navbar?
> 
> Was this fixed?

I believe it was, but if it was not request for that to be changed.

**Comment by toni-santos** [comment]
> > > There is no way for a user to go to the shop unless they know the exact Url, maybe a link in the Navbar?
> > 
> > 
> > Was this fixed?
> 
> I believe it was, but if it was not request for that to be changed.

This hasn't been addressed has #129 introduces an altered navbar and hasn't been merged.

**Review by pedronunes19 — APPROVED** [review]

**Pull request merged**